### DISCUSSION TITLE
/usr/bin/curl: can't load library 'libssl.so.3'

### DISCUSSION THREAD

**Discussion by jwgn** [Freetz-NG/freetz-ng#838] [Allgemein] [answered]
Ich habe curl installiert, es lässt sich aber nicht starten.

Vermutlich, weil unter Shared libraries -> Crypto & SSL die `OpenSSL SSL/TLS library (libssl.to)` nicht automatisch ausgewählt wird, wenn man curl auswählt?

**Comment by fda77** [comment]
Wie kommst du auf diese Idee! Manchmal braucht man das nicht?

**Comment by jwgn** [comment]
Der curl Fehler lautet, er kann die library nicht laden. Daraus schließe ich, dass er die library braucht? Nicht?

**Comment by fda77** [comment]
Nur du weisst was du gemacht hast?

**Comment by jwgn** [comment]
Das habe ich gemacht

![WindowsTerminal_bODAF7ZTaH](https://github.com/Freetz-NG/freetz-ng/assets/15985023/c15e6558-3f93-4888-899e-2e3ec7442c65)

**Reply by fda77** [reply]
Ja....
Weisst du was mich am meisten langweilt? Immer und immer wieder die gleichen Fragen zu stellen!
Hast du dir mal überlegt wie viele openssl versionen es in freetz gibt. compiler, libc etc

**Comment by jwgn** [comment]
Was soll denn die unfreundliche Antwort? Ich habe darauf hingewiesen, dass curl nicht funktioniert, wenn es als Paket ausgewählt wird, weil eine library fehlt. Und eine Vermutung geäußert, woran es liegen könnte. Mach was aus der Info oder eben nicht.

**Reply by aus-sua** [reply]
Ich erkläre Dir es einmal sehr freundlich und kurz:

Zum 1000-ten mal schlägt hier ein lustiger Zeitgenosse auf, der angibt, einen
Fehler in freetz-NG gefunden zu haben...
Ja, das ist gut möglich, es stecken sicherlich auch noch mehrere ungefundene
Fehler in freetz/freetz-NG.

Nur,
es ist dabei aber DEINE ganz einfache und ELEMENTARE Aufgabe, zumindest die
NÖTIGSTEN Angaben zur Fehlersuche MITZULIEFERN, denn das ist seit
ca. 50 Jahren der Standard zur Fehlerbeseitigung (*) in einer OS-Software!

Dazu gehören z.B.:
- die Angaben zur Fritzbox  (Typ + Version FOS), für die Du eine freetz-OS erstellst hast
- die Fehlermeldung und die Beschreibung des Auftretens
- die Commits-Version des freetz-NG
- DEINE ~~.config.compressed-Datei~~ .config-Datei der freetz-OS-Erstellung
- die fmake-Datei DEINER freetz-OS-Erstellung
- die Version/Typ DEINES Betriebsystems, mit dem DU die freetz-OS erstellst hast

usw. usw.
Liebend gerne kannst Du auch mehr Infos liefern oder, dann noch besser
(quasi der freundlichste Goldstandard), einen Pull requests mit der Fehlerbehebung
liefern.


(*)
freetz/freetz-NG ist ein Baukasten-System zur OS-Erstellung, bei dem es 100.000-te von
Abhängigkeiten der zu erstellenden Programmversionen gibt.
Erwartest DU ernsthaft, der Collaborator benutzt eine "Glaskugel" um den Fehler
zu suchen/finden, weil er es leid ist, ständig die NICHT gelieferten Fehlerinfos immer und
immer wieder abzufragen?

Nun ja,
und nun:
Mach DU was aus den Infos oder eben nicht.

**Comment by jwgn** [comment]
Ich habe keinen Fehler gemeldet. Ich habe eine Frage gestellt. Seit wann ist das hier ein Bugtracker? Da bei mir steht da `question in [Feedback]` und nicht Bugreport.

Danke für die Diskussion. Schönen Abend noch.

**Comment by aus-sua** [comment]
Ach so...
Nur ändert das aber nichts an der Tatsache:

Selbst "nur" zum beantworten Deiner "Frage" fehlen die o.g. Infos.

Wenn Du zu faul oder zu cool bist, diese mit zu liefern, dann kann Dir
eben beim besten und freundlichsten Willen da keiner weiterhelfen
(warum: s.o., es fehlt an einer funktionierenden Glaskugel...).

**Comment by jwgn** [comment, marked as answer]
Falls hier noch jemand drüber stolpert: Es ist genau so, wie im ersten Post vermutet. Ich habe das Image neu gebaut und die einzige Änderung, die ich vorgenommen habe, war, `OpenSSL SSL/TLS library (libssl.to)` zusätzlich manuell auszuwählen. Jetzt funktioniert curl.

> Wie kommst du auf diese Idee! Manchmal braucht man das nicht?

War einfach nur eine Idee! Offenbar braucht man das manchmal schon?

> Wenn Du zu faul oder zu cool bist, diese mit zu liefern, dann kann Dir
> eben beim besten und freundlichsten Willen da keiner weiterhelfen
> (warum: s.o., es fehlt an einer funktionierenden Glaskugel...).

Offenbar habe ich gar keine Hilfe benötigt, sondern nur auf ein Problem hingewiesen und die Lösung gleich mitgeliefert. Ganz ohne Glaskugel.

**Comment by fda77** [comment]
> und die einzige Änderung, die ich vorgenommen habe, [..]. Jetzt funktioniert curl.

Und wie kommt dann Curl ins Image? Das ist nicht ausgewählt, als gab es wohl noch andere Änderungen...

> (libssl.to) zusätzlich manuell auszuwählen

Es kann schon sein dass diese Auswahl irgendwo fehlt, aus dem was du geschrieben hast werd ich aber nicht schlau und wüsste nicht wo was zu ändern wäre

**Comment by jwgn** [comment]
Ich meinte die einzige Änderung zu dem vorherigen Zustand. Natürlich habe ich viele Änderungen vorgenommen. Z.B. curl zur Installation ausgewählt inkl SSL Suport. Siehe Screenshot oben.

Das führt aber eben dazu, dass curl beim Start die libssl.to vermisst. Wenn man `OpenSSL SSL/TLS library (libssl.to)` mitinstalliert, dann funktioniert curl.

Also wenn man `Curl`  und `Select SSL Support`, installiert, muss `OpenSSL SSL/TLS library (libssl.to)` als Abhängigkeit mitkommen.
